include/engine/routing_algorithms/many_to_many.hpp

```diff
@@ -16,24 +16,45 @@ namespace engine
 namespace routing_algorithms
 {
 
+namespace
+{
+struct NodeBucket
```

Comment:
This seems to be an MLD only structure but part of a general many_to_many implementation. Any way wie could connect it with MLD?

Reply:
it is also used in CH. I just moved `NodeBucket` declaration  to a common place

Reply:
It specifically refers to a column in MLD in the commentary? Is that just a misnomor?

Reply:
no, it is always a column index 

At https://github.com/Project-OSRM/osrm-backend/blob/4b8f85115153755ee55b5e72183611d7362f0e6c/src/engine/routing_algorithms/many_to_many_mld.cpp#L404 it is used as a column index in column-major matrix that is in the same time  a transposed matrix if treated as a row-major matrix. I just moved a transpose operation at that line, otherwise an explicit transpose of returned matrix is required at https://github.com/Project-OSRM/osrm-backend/blob/4b8f85115153755ee55b5e72183611d7362f0e6c/src/engine/routing_algorithms/many_to_many_mld.cpp#L556-L557

Reply:
ah. sorry. miss-read this as column-index in the MLD matrix, not the result matrix. All good here.